src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/RazorFormattingOptions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Razor.Formatting;
+
+internal sealed record RazorFormattingOptions
+{
+    public bool InsertSpaces { get; init; }
+    public int TabSize { get; init; }
+    public bool CodeBlockBraceOnNextLine { get; init; }
```

Comment:
I kinda like not having to specify all options, but of course I realised that I stuffed up the default instance, so I've fixed that. I've copied the style from Roslyn ([for example](https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/AutoFormattingOptions.cs)), except moved the static to the top since I'm not an animal. Let me know if you feel strongly about constructors vs initializers

Reply:
At the very least, I would expect `InsertSpaces` and `TabSize` to be specified every time. If that's true, then maybe you could just add a default value to the `CodeBlockBraceOnNextLine` parameter.

All of that said, I'm definitely sympathetic to non-positional records. You lose deconstruction, but you also lose the friction of having to update constructors when you need to add more properties in the future. That's totally a judgement call and I'm happy either way!

Reply:
That friction was my main motivation for this type existing in the first place. Plumbing through _another_ flag on `FormattingContext` was what I started having to do, which led to the creation of this type. Also once Alex's PR is in, there is another flag to come here for FUSE. Though I will admit, having the last parameter be optional, and the new FUSE one, would probably be just as good, as they're only needed for full document formatting, so only one place would have to pay the price.

I'll think about it some more. Not merging this until Alex's PR is in anyway.